core/src/main/java/org/apache/carbondata/core/index/Segment.java

```diff
@@ -378,11 +382,10 @@ public void write(DataOutput out) throws IOException {
       out.writeUTF(segmentString);
     }
     out.writeLong(indexSize);
-    if (segmentPath == null) {
-      out.writeBoolean(false);
-    } else {
+    if (isExternalSegment()) {
```

Comment:
No need to have if check. Just "out..writeBoolean(isExternalSegment)"

Reply:
Done